management/clipper_manager.py

```diff
@@ -0,0 +1,526 @@
+                ]
+        input_type : str
+            One of "integers", "floats", "doubles", "bytes", or "strings".
+        selection_policy : str
+            The name of the model selection policy to be used for the
+            application.
+        output_type : str, optional
+            Either "double" or "int". Default is "double".
+        slo_micros : int, optional
+            The query latency objective for the application in microseconds.
+            Default is 20,000 (20 ms).
+        """
+        url = "http://%s:1338/admin/add_app" % self.host
+        req_json = json.dumps({
+            "name": name,
+            "candidate_models": candidate_models,
+            "input_type": input_type,
+            "output_type": output_type,
+            "selection_policy": selection_policy,
+            "latency_slo_micros": slo_micros
+        })
+        headers = {'Content-type': 'application/json'}
+        r = requests.post(url, headers=headers, data=req_json)
+        print(r.text)
+
+    def list_apps(self):
+        """List the names of all applications registered with Clipper.
+
+        Returns
+        -------
+        str
+            The string describing each registered application. If no
+            applications are found, an empty string is returned.
+        """
+        with hide("output", "running"):
+            result = local(("redis-cli -h {host} -p 6379 -n {db} keys \"*\""
+                            .format(host=self.host,
+                                    db=REDIS_APPLICATION_DB_NUM)),
+                           capture=True)
+
+            if len(result.stdout) > 0:
+                return result.stdout
+            else:
+                print("Clipper has no applications registered")
+                return ""
+
+    def get_app_info(self, name):
+        """Gets detailed information about a registered application.
+
+        Parameters
+        ----------
+        name : str
+            The name of the application to look up
+
+        Returns
+        -------
+        dict or None
+            Returns a dict with the application info if found. If the application
+            is not registered, None is returned.
+        """
+        with hide("output", "running"):
+            result = local("redis-cli -h {host} -p 6379 -n {db} hgetall {name}".format(
+                host=self.host, name=name, db=REDIS_APPLICATION_DB_NUM), capture=True)
+
+            if len(result.stdout) > 0:
+                splits = result.stdout.split("\n")
+                fmt_result = dict([(splits[i], splits[i+1])
+                                   for i in range(0, len(splits), 2)])
+                pp = pprint.PrettyPrinter(indent=2)
+                pp.pprint(fmt_result)
+                return fmt_result
+            else:
+                warn("Application \"%s\" not found" % name)
+                return None
+
+    def inspect_selection_policy(self, app_name):
```

Comment:
The parameter `uid` that is specified in the documentation is not included in the method prototype.